src/main/java/io/openliberty/tools/common/plugins/util/VariableUtility.java

```diff
@@ -90,35 +89,58 @@ public static String resolveVariables(CommonLoggerI log, String nodeValue, Colle
         String value = null;
-        if(!libDirPropFiles.containsKey(propertyName)) {
-            value = props.getProperty(propertyName);
-            if (value == null) {
-                // Check for default value since no other value found.
-                value = defaultProps.getProperty(propertyName);
-            }
-            if (value == null && propertyName.startsWith("env.") && propertyName.length() > 4) {
-                // Look for property without the 'env.' prefix
-                String newPropName = propertyName.substring(4);
-                value = props.getProperty(newPropName);
-                if (value == null) {
-                    // Check for default value since no other value found.
-                    value = defaultProps.getProperty(newPropName);
-                }
-            }
-        } else {
-            File envDirectory = libDirPropFiles.get(propertyName);
-            value = envDirectory.toString();
-        }  
-        
-        if (value != null && value.startsWith("\"") && value.endsWith("\"")) {
-            // need to remove beginning/ending quotes
-            if (value.length() > 2) {
-                value = value.substring(1, value.length() -1);
+        if (libertyDirPropFiles.containsKey(propertyName)) {
+            return stripQuotes(libertyDirPropFiles.get(propertyName).toString());
+        }
+
+        value = lookupProperty(prop, defaultProps, propertyName);
+        if (value != null) {
+            return value;
+        }
+
+        // try again with non-alphanumeric values replaced with '_', which is exactly \W in regex
+        String propertyNameVariation = propertyName.replaceAll("\\W", "_");
+        value = lookupProperty(prop, defaultProps, propertyNameVariation);
+        if (value != null) {
+            return value;
+        }
+
+        // try again with propertyNameVariation.toUpperCase()
+        propertyNameVariation = propertyNameVariation.toUpperCase();
+        value = lookupProperty(prop, defaultProps, propertyNameVariation);
+        if (value != null) {
+            return value;
+        }
+
+        // support for versions <19.0.0.3. Look for property without the 'env.' prefix
+        if (propertyName != null && propertyName.startsWith("env.") && propertyName.length() > 4) {
+            value = lookupProperty(prop, defaultProps, propertyName.substring(4));
+            if (value != null) {
+                return value;
             }
         }
+
+        return value; 
+    }
+
+    private static String stripQuotes(String value) {
+        if (value.startsWith("\"") && value.endsWith("\"") && value.length() > 2) {
+            return value.substring(1, value.length() - 1);
+        }
         return value;
     }
-    
+
+    private static String lookupProperty(Properties prop, Properties defaultProps, String propertyName) {
+        if (prop.containsKey(propertyName)) {
+            return stripQuotes(prop.getProperty(propertyName));
```

Comment:
Just because `prop` contains the key does not mean the value from `getProperty` is not null. I think it would be best to guard against a NPE in `stripQuotes` before calling methods on `value`.